variables.tf

```diff
@@ -223,6 +223,18 @@ variable "origin_request_policy_id" {
     EOT
 }
 
+variable "origin_shield" {
+  type = object({
+    enabled = bool
+    region  = string
+  })
+  description = <<-EOT
+    The CloudFront Origin Shield settings.
+    If `region` is set to `auto` it will determine the closest region where Origin Shield is available.
```

Comment:
This is really clever, but I've never seen this type of behaviour in any of our modules. I'm not sure if maybe this is an exception, but I definitely am going to need someone else's opinion on it.

In my opinion, it's a little too "smart" for a Terraform configuration.

Reply:
I think it's acceptable, especially if it doesn't change to often and it's easy to add other regions as they come online.  

My main suggestion though is we should move the `local.origin_shield_region_mapping` to a `var. origin_shield_region_mapping` with a default of this map so that anyone can change the behavior without forking the module.

Reply:
Good suggestion. I have changed the `local` into a `var` so anyone can override its behavior.